modules/cli/src/main/java/org/apache/ignite/cli/spec/ClusterCommandSpec.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.cli.spec;
+
+import java.util.ArrayList;
+import java.util.List;
+import javax.inject.Inject;
+import org.apache.ignite.cli.builtins.cluster.ClusterApiClient;
+import picocli.CommandLine;
+import picocli.CommandLine.Option;
+
+/**
+ * Commands for managing Ignite cluster as a whole.
+ */
+@CommandLine.Command(
+        name = "cluster",
+        description = "Manages an Ignite cluster as a whole.",
+        subcommands = {
+                ClusterCommandSpec.InitClusterCommandSpec.class,
+        }
+)
+public class ClusterCommandSpec extends CategorySpec {
+    /**
+     * Initializes an Ignite cluster.
+     */
+    @CommandLine.Command(name = "init", description = "Initializes an Ignite cluster.")
+    public static class InitClusterCommandSpec extends CommandSpec {
+
+        @Inject
+        private ClusterApiClient clusterApiClient;
+
+        /**
+         * Address of the REST endpoint of the receiving node in host:port format.
+         */
+        @Option(names = "--node-endpoint", required = true,
+                description = "Address of the REST endpoint of the receiving node in host:port format")
+        private String nodeEndpoint;
```

Comment:
Should you use `NodeEndpointOptions` here?

Reply:
`NodeEndpointOptions` allows the endpoint to be omitted. In the case of the init command, it seems super-important to make sure that the user knows exactly what endpoint they use, so I made it mandatory, hence `NodeEndpointOptions` cannot be used.